ibmwatson.class.nut

```
@@ -0,0 +1,263 @@
+        req.sendasync(function(res) {
+            _processResponse(res, cb);
+        }.bindenv(this));
+    }
+
+    /***************************************************************************************
+     * formatTimestamp
+     * Returns: time formatted as "2015-12-03T00:54:51Z"
+     * Parameters:
+     *      ts (optional) : integer - epoch timestamp
+     **************************************************************************************/
+    function formatTimestamp(ts = null) {
+        local d = ts ? date(ts) : date();
+        return format("%04d-%02d-%02dT%02d:%02d:%02dZ", d.year, d.month+1, d.day, d.hour, d.min, d.sec);
+    }
+
+    /***************************************************************************************
+     * _createHeaders
+     * Returns: request header table
+     * Parameters:
+     *      headers (optional) : table - additional headers
+     **************************************************************************************/
+    function _createHeaders(headers = {}) {
+        local auth = http.base64encode(format("%s:%s", _apiKey, _authToken));
+        if (!("Authorization" in headers)) headers["Authorization"] <- format("Basic %s", auth);
+        if (!("Content-Type" in headers)) headers["Content-Type"] <- "application/json";
+        return headers;
+    }
+
+    /***************************************************************************************
+     * _processResponse
+     * Returns: null
+     * Parameters:
+     *      res : the response object from Watson
+     *      cb : the callback function passed into the request or null
+     **************************************************************************************/
+    function _processResponse(res, cb) {
+        local status = res.statuscode;
+        local err = (status < 200 || status >= 300) ? _getError(status) : null;
+        try {
+            res.body = (res.body == "") ? {} : http.jsondecode(res.body);
```

Imagine you have 5xxx response and an invalid response body that will result in JSON parser error here. So we'll actually hide the original error for the user here. Do you think this can be an issue?